graylog2-shared/src/main/java/org/graylog2/shared/security/ldap/LdapEntry.java

```diff
@@ -41,6 +54,13 @@ public String put(String key, String value) {
         return attributes.put(key.toLowerCase(), value);
     }
 
+    public void addGroups(Collection<String> groups) {
```

Comment:
LdapEntry is just a glorified holder for user attributes pulled out of ldap. It will only be used to later create the actual user entry in the collection.
I'm not sure it's worth making it immutable given that it is only used once.